server/services/v1/cache/query_runner.go

```diff
@@ -0,0 +1,312 @@
+		BaseQueryRunner: NewBaseQueryRunner(f.encoder, accessToken, f.cacheStore),
+		req:             r,
+	}
+}
+
+func (runner *CreateCacheQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (*CacheResponse, context.Context, error) {
+	currentSub, err := request.GetCurrentSub(ctx)
+	if err != nil && config.DefaultConfig.Auth.Enabled {
+		return nil, ctx, errors.Internal("Failed to get current sub for the request")
+	}
+
+	_, err = tenant.CreateCache(ctx, tx, runner.req.GetProject(), runner.req.GetName(), currentSub)
+	if err != nil {
+		return nil, ctx, err
+	}
+	return &CacheResponse{
+		Status: database.CreatedStatus,
+	}, ctx, nil
+}
+
+func (runner *DeleteCacheQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (*CacheResponse, context.Context, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, ctx, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, ctx, err
+	}
+	internalKeys, err := runner.cacheStore.Keys(ctx, encodedCacheTableName, "*")
+	if err != nil {
+		return nil, ctx, err
+	}
+	for _, internalKey := range internalKeys {
+		// translate the key to user key
+		userKey := cacheKeyInternalToExternal(internalKey)
+		_, err = runner.cacheStore.Delete(ctx, encodedCacheTableName, userKey)
+		if err != nil {
+			log.Warn().Str("cacheTableName", encodedCacheTableName).Str("cacheKey", userKey).Msg("Failed to delete cache key")
+		}
+	}
+
+	_, err = tenant.DeleteCache(ctx, tx, runner.req.GetProject(), runner.req.GetName())
+	if err != nil {
+		log.Warn().
+			Str("tenant", tenant.GetNamespace().StrId()).
+			Str("project", runner.req.GetProject()).
+			Str("cache", runner.req.GetName()).
+			Msg("Failed to update project metadata entry to delete cache")
+		return nil, ctx, err
+	}
+	return &CacheResponse{
+		Status: database.DeletedStatus,
+	}, ctx, nil
+}
+
+func (runner *SetQueryRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	options := &cache.SetOptions{
+		NX: runner.req.GetNx(),
+		XX: runner.req.GetXx(),
+		EX: time.Duration(runner.req.GetEx()) * time.Second,
+	}
+	err = runner.cacheStore.Set(ctx, encodedCacheTableName, runner.req.GetKey(), internal.NewCacheData(runner.req.GetValue()), options)
+
+	if err != nil {
+		return nil, errors.Internal("Failed to invoke set, reason %s", err.Error())
+	}
+	return &CacheResponse{
+		Status: database.InsertedStatus,
```

Comment:
I think it is fine to Cache defining its own meaningful statutes messages.

Reply:
sure. created new statuses.